torch/_functorch/_aot_autograd/autograd_cache.py

```diff
@@ -3,19 +3,79 @@
 """
 from __future__ import annotations
 
+import functools
 import logging
-from typing import TYPE_CHECKING
+import os
 
-from torch._inductor.codecache import _ident, FxGraphCachePickler
+import torch
+from torch._inductor.codecache import (
+    _ident,
+    FxGraphCachePickler,
+    get_code_hash,
+    get_inductor_root,
+)
+from torch.fx.node import Node
 
 from .schemas import AOTConfig  # noqa: F401
 
-if TYPE_CHECKING:
-    import torch
-
 log = logging.getLogger(__name__)
 
 
+class BypassAOTAutogradCache(Exception):
+    pass
+
+
+def check_node_safe(node: Node):
+    """
+    Checks that the node only uses supported operators.
+    """
+
+    def is_torch_function(target):
+        return type(target).__name__.startswith("builtin")
```

Comment:
hmm why does the builtin check map to torch functions? Do you also need to check that the function's module is "torch"?

It might also be worth a comment about the ways in which is wrong plus the "ideal" state:

(1) ways it is wrong: if a user monkey-patches a torch.* API, so we can't tell

(2) ideal state: if dynamo inlined every single torch API all the way to its C binding, we could check if the type of the fn is `builtin_function_or_method`(representing a C-bound python fn that we know is not monkey-patched)

Reply:
Oops, your right, need to also check the module haha